MediaBrowser.LocalMetadata/Images/LocalImageProvider.cs

```diff
@@ -449,32 +449,38 @@ private bool AddImage(List<FileSystemMetadata> files, List<LocalImageInfo> image
             return added;
         }
 
-        private bool AddImage(List<FileSystemMetadata> files, List<LocalImageInfo> images, string name, ImageType type)
+        private static bool AddImage(IReadOnlyList<FileSystemMetadata> files, List<LocalImageInfo> images, string name, ImageType type, string? prefix = null)
         {
-            var image = GetImage(files, name);
+            var image = GetImage(files, name, prefix);
 
-            if (image != null)
+            if (image == null)
             {
-                images.Add(new LocalImageInfo
-                {
-                    FileInfo = image,
-                    Type = type
-                });
-
-                return true;
+                return false;
             }
 
-            return false;
+            images.Add(new LocalImageInfo
+            {
+                FileInfo = image,
+                Type = type
+            });
+
+            return true;
         }
 
-        private static FileSystemMetadata? GetImage(IReadOnlyList<FileSystemMetadata> files, string name)
+        private static FileSystemMetadata? GetImage(IReadOnlyList<FileSystemMetadata> files, string name, string? prefix = null)
         {
+            var fileNameLength = name.Length + (prefix?.Length ?? 0);
             for (var i = 0; i < files.Count; i++)
             {
                 var file = files[i];
-                if (!file.IsDirectory
-                    && file.Length > 0
-                    && Path.GetFileNameWithoutExtension(file.FullName.AsSpan()).Equals(name, StringComparison.OrdinalIgnoreCase))
+                if (file.IsDirectory || file.Length <= 0 || file.FullName.Length != fileNameLength)
```

Comment:
This won't work?
`file.FullName` return the whole path, not just the file name.

Reply:
Right, brainfart.